crates/storage/provider/src/providers/snapshot.rs

```diff
@@ -0,0 +1,205 @@
+use crate::HeaderProvider;
+use reth_db::{
+    table::{Decompress, Table},
+    HeaderTD,
+};
+use reth_interfaces::RethResult;
+use reth_nippy_jar::{NippyJar, NippyJarCursor};
+use reth_primitives::{BlockHash, BlockNumber, Header, SealedHeader, U256};
+use std::ops::RangeBounds;
+
+/// SnapshotProvider
+///
+///  WIP Rudimentary impl just for testes
+/// TODO: should be able to walk through snapshot files/block_ranges
+/// TODO: Arc over NippyJars and/or NippyJarCursors (LRU)
+#[derive(Debug)]
+pub struct SnapshotProvider<'a> {
+    /// NippyJar
+    pub jar: &'a NippyJar,
+}
+
+impl<'a> SnapshotProvider<'a> {
+    fn cursor(&self) -> NippyJarCursor<'a> {
+        NippyJarCursor::new(self.jar, None).unwrap()
+    }
+}
+
+impl<'a> HeaderProvider for SnapshotProvider<'a> {
+    fn header(&self, block_hash: &BlockHash) -> RethResult<Option<Header>> {
+        // WIP
+        let mut cursor = self.cursor();
+
+        let header = Header::decompress(
+            &cursor.row_by_key_with_cols::<0b01, 2>(&block_hash.0).unwrap().unwrap()[0],
```

Comment:
let's introduce constants for masks straight away, otherwise it makes it hard to understand what's going on